### ISSUE TITLE
 Parameter propagation when duplicating component broken in OMEdit 1.22.1

### ISSUE THREAD

**Issue by pierre-haessig** [OpenModelica/OpenModelica#11790]
Hi,

Here is a sort of follow-up of #11695 and #11595 which got fixed in  OMEdit 1.22.1 (by #11599 and #11655). Now it's a variant: for a component which contains a parameter set to a named parameter (rather than a numerical litteral), the duplication of that component (Ctrl-D in the Diagram View) fails with an error message and the parameter of the duplicated component isn't set.

### Steps to Reproduce

1. Load the following model in OMEdit 1.22.1:

```Modelica
model ParameterPropagation "Bug in duplication of parameter propagation in OM 1.22.1"
  parameter Real Ccircuit;
  Modelica.Electrical.Analog.Basic.Capacitor capacitor(C = Ccircuit)  annotation(
    Placement(transformation(origin = {8, 14}, extent = {{-10, -10}, {10, 10}})));
equation

annotation(
    uses(Modelica(version = "4.0.0")));
end ParameterPropagation;
```

2. In the diagram view, duplicate the capacitor → creates a `capacitor1` component, but with error message and its `C` parameter **not set**. Error message is:

```
[8] 18:11:51 Scripting Error
Unable to set the element modifier value using command setElementModifierValue(ParameterPropagationDuplication, capacitor1, $Code(((C = Ccircuit))))
```

### Expected Behavior

The duplicated component should have its parameters duplicated as well (`capacitor1(C = Ccircuit)`)

### Version and OS

 - OpenModelica Version: [e.g. `omc --version` or `Help->About OMEdit` from OMEdit]
 - OS: Debian 12 stable

**Comment by casella** [comment]
@perost is this an API issue or an OMEdit issue?

**Comment by perost** [comment]
> @perost is this an API issue or an OMEdit issue?

I think it's an OMEdit issue, there are too many parentheses in the `setElementModifierValue` call. It should be `$Code((C = Ccircuit))` and not `$Code(((C = Ccircuit)))`, otherwise it's a syntax error (the part inside the `$Code()` is parsed as a modifier, so it would be equivalent to trying to set `capacitor((C = Ccircuit))`).

**Comment by casella** [comment]
@adeas31 can you please take care of that?

**Comment by adeas31** [comment]
Fixed with #12046 and ported to maintenance 1.22 with #12050
